sdk/servicebus/azure-servicebus/tests/async_tests/test_queues_async.py

```diff
@@ -65,990 +67,1164 @@ def print_message(message):
-    queue_client = QueueClient.from_connection_string(
-        live_servicebus_config['conn_str'],
-        name=standard_queue,
-        debug=False)
-    queue_client.get_properties()
-
-    async with queue_client.get_sender() as sender:
-        for i in range(10):
-            message = Message("Handler message no. {}".format(i))
-            message.enqueue_sequence_number = i
-            await sender.send(message)
-
-    messages = []
-    receiver = queue_client.get_receiver(mode=ReceiveSettleMode.ReceiveAndDelete, idle_timeout=5)
-    async for message in receiver:
-        messages.append(message)
-        with pytest.raises(MessageAlreadySettled):
-            await message.complete()
-
-    assert not receiver.running
-    assert len(messages) == 10
-    time.sleep(30)
-
-    messages = []
-    receiver = queue_client.get_receiver(mode=ReceiveSettleMode.ReceiveAndDelete, idle_timeout=5)
-    async for message in receiver:
-        messages.append(message)
-    assert len(messages) == 0
-
-@pytest.mark.liveTest
-@pytest.mark.asyncio
-async def test_async_queue_by_queue_client_conn_str_receive_handler_with_stop(live_servicebus_config, standard_queue):
-    queue_client = QueueClient.from_connection_string(
-        live_servicebus_config['conn_str'],
-        name=standard_queue,
-        debug=False)
-
-    async with queue_client.get_sender() as sender:
-        for i in range(10):
-            message = Message("Stop message no. {}".format(i))
-            await sender.send(message)
-
-    messages = []
-    receiver = queue_client.get_receiver(idle_timeout=5)
-    async for message in receiver:
-        messages.append(message)
-        await message.complete()
-        if len(messages) >= 5:
-            break
-
-    assert receiver.running
-    assert len(messages) == 5
-
-    async with receiver:
-        async for message in receiver:
-            messages.append(message)
-            await message.complete()
-            if len(messages) >= 5:
-                break
 
-    assert not receiver.running
-    assert len(messages) == 6
+class ServiceBusQueueTests(AzureMgmtTestCase):
 
-@pytest.mark.liveTest
-@pytest.mark.asyncio
-async def test_async_queue_by_servicebus_client_iter_messages_simple(live_servicebus_config, standard_queue):
-    client = ServiceBusClient(
-        service_namespace=live_servicebus_config['hostname'],
-        shared_access_key_name=live_servicebus_config['key_name'],
-        shared_access_key_value=live_servicebus_config['access_key'],
-        debug=False)
-
-    queue_client = client.get_queue(standard_queue)
-    async with queue_client.get_receiver(idle_timeout=5, mode=ReceiveSettleMode.PeekLock) as receiver:
+    @pytest.mark.liveTest
+    @pytest.mark.skipif(not AreLiveTestsEnabled(), reason="This test only runs against live resources")
+    @pytest.mark.asyncio
+    @ResourceGroupPreparer(name_prefix='servicebustest')
```

Comment:
Please pass `random_name_enabled=True` to all your ResourceGroupPreparer, that's the value I want to make default once I have re-recorded all Mgmt tests correctly. New PRs should always pass this flag, and new preparer implementation should force it and not even expose it

Reply:
I put the partial within devtools_testutils to keep that logic centralized and make swapping out trivial.  Don't hesitate to shout if this is problematic.